tests/cupy_tests/linalg_tests/test_norms.py

```diff
@@ -206,3 +206,120 @@ def test_slogdet_one_dim(self, dtype):
             a = testing.shaped_arange((2,), xp, dtype)
             with pytest.raises(numpy.linalg.LinAlgError):
                 xp.linalg.slogdet(a)
+
+
+@testing.parameterize(
+    *testing.product({"ord": [-numpy.inf, -2, -1, 1, 2, numpy.inf, "fro"]})
+)
+class TestCond(unittest.TestCase):
+    def test_basic_nonsvd(self):
+        # Smoketest the non-svd norms
+        A = cupy.array([[1.0, 0, 1], [0, -2.0, 0], [0, 0, 3.0]])
+        testing.assert_array_almost_equal(cupy.linalg.cond(A, cupy.inf), 4)
+        testing.assert_array_almost_equal(
+            cupy.linalg.cond(A, -cupy.inf), 2 / 3
+        )
+        testing.assert_array_almost_equal(cupy.linalg.cond(A, 1), 4)
+        testing.assert_array_almost_equal(cupy.linalg.cond(A, -1), 0.5)
+        testing.assert_array_almost_equal(
+            cupy.linalg.cond(A, "fro"), cupy.sqrt(265 / 12)
+        )
```

Comment:
Could you split this function into a separate class? This is unnecessarily executed multiple times (for each `testing.parameterize`).